pkg/services/store/entity/sqlstash/sql_storage_server.go

```diff
@@ -1238,21 +1260,15 @@ func (s *sqlEntityServer) List(ctx context.Context, r *entity.EntityListRequest)
 		return nil, err
 	}
 	defer func() { _ = rows.Close() }()
-	rsp := &entity.EntityListResponse{
-		ResourceVersion: s.snowflake.Generate().Int64(),
-	}
 	for rows.Next() {
 		result, err := rowToEntity(rows, r)
 		if err != nil {
 			return rsp, err
 		}
 
 		// found more than requested
-		if int64(len(rsp.Results)) >= entityQuery.limit {
-			continueToken := &ContinueToken{
-				Sort:        r.Sort,
-				StartOffset: entityQuery.offset + entityQuery.limit,
-			}
+		if entityQuery.limit > 0 && int64(len(rsp.Results)) >= entityQuery.limit {
+			continueToken.StartOffset = entityQuery.offset + entityQuery.limit
```

Comment:
This is a question about offset and ordering.

If an element is updated between two pages, will the offset still work ? 

Let's pretend we the user isn't sorting.
- When computing the initial page, the count + max_rv was only looking at data from the entity table
- When computing the next page, the data is joining the entity table with the entity_history table.
Given the dataset is slightly different, will offset actually work ?

Reply:
As long as the 2 tables are in sync the offsets will match for a given RV because the query against entity_history will return the same data as existed in entity at the specified RV.